resolvers/npm_cache/tarball_extract.rs

```diff
@@ -199,6 +199,34 @@ pub enum ExtractTarballError {
   #[error(transparent)]
   Io(#[from] std::io::Error),
   #[class(generic)]
+  #[error("Failed creating '{}'", path.display())]
+  FailedCreatingDir {
+    path: PathBuf,
+    #[source]
+    source: std::io::Error,
+  },
+  #[class(generic)]
+  #[error("Failed canonicalizing '{}'", path.display())]
+  FailedCanonicalizing {
+    path: PathBuf,
+    #[source]
+    source: std::io::Error,
+  },
+  #[class(generic)]
+  #[error("Failed opening '{}'", path.display())]
+  FailedOpening {
+    path: PathBuf,
+    #[source]
+    source: std::io::Error,
+  },
+  #[class(generic)]
+  #[error("Failed writing '{}'", path.display())]
+  FailedWriting {
+    path: PathBuf,
+    #[source]
+    source: std::io::Error,
+  },
```

Comment:
Kind of feel like we should just have a generic IO error type with a context and what kind of operation failed. Might be good for deno_error.